src/lib/io/sol-efivarfs-storage.c

```diff
@@ -214,11 +214,7 @@ sol_efivars_read_raw(const char *name, struct sol_buffer *buffer)
         goto end;
     }
 
-    if (buffer->capacity) {
-        r = sol_util_fill_buffer_exactly(fd, buffer, buffer->capacity);
-    } else {
-        r = sol_util_load_file_fd_buffer(fd, buffer);
-    }
+    r = sol_util_load_file_fd_buffer(fd, buffer);
```

Comment:
Yeah, to accommodate the case where they may lie on a bit field. On my sample I made sure this case was tested:
```
    {
     "name": "bool_test",
     "offset": 3,
     "size": 4,
     "bit_size": 1,
     "bit_offset": 2
    },
```,

a 4-byte size reserved space where the only relevant data is one bit at (bit) offset 2. We must read the whole 4-byte chunk, regardless of the position of the useful data in it (we extract it later).